bonsai/BudgetTransactions/BudgetTransactions.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  BudgetTransactions.swift
+//  bonsai
+//
+//  Created by hoang on 08.12.2021.
+//
+
+import SwiftUI
+
+struct VisualEffectView: UIViewRepresentable {
+    var effect: UIVisualEffect?
+    func makeUIView(context: UIViewRepresentableContext<Self>) -> UIVisualEffectView { UIVisualEffectView() }
+    func updateUIView(_ uiView: UIVisualEffectView, context: UIViewRepresentableContext<Self>) { uiView.effect = effect }
+}
+
+struct RoundedCorner: Shape {
+
+    var radius: CGFloat = .infinity
+    var corners: UIRectCorner = .allCorners
+
+    func path(in rect: CGRect) -> Path {
+        let path = UIBezierPath(roundedRect: rect, byRoundingCorners: corners, cornerRadii: CGSize(width: radius, height: radius))
+        return Path(path.cgPath)
+    }
+}
+
+extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
+        clipShape(RoundedCorner(radius: radius, corners: corners))
+    }
+}
+
+
+struct BudgetTransactions: View {
+    
+    private typealias TransactionGroup = [String: [Transaction]]
+    private let transactionsByDate: TransactionGroup
+    
+    init(transactions: [Transaction]) {
+        var dict: TransactionGroup = [:]
+        for transaction in transactions {
+            let date = Date.dateMapper(date: transaction.date)
+            if dict.keys.contains(date) {
+                if var arr = dict[date] {
+                    arr.append(transaction)
+                    dict[date] = arr
+                }
+            } else {
+                dict[date] = [transaction]
+            }
+        }
+        transactionsByDate = dict
+    }
+        
+    var body: some View {
+        ZStack {
+            VisualEffectView(
+                effect: UIBlurEffect(style: .dark))
+                .edgesIgnoringSafeArea(.all)
+            
+            VStack(alignment: .leading) {
+                HStack {
+                    Spacer()
+                    RoundedCorner()
+                        .foregroundColor(BonsaiColor.disabled)
+                        .frame(width: 60, height: 5, alignment: .center)
+                        .cornerRadius(25, corners: .allCorners)
+                    Spacer()
+                }
+                
+                Text("Transactions")
+                    .font(BonsaiFont.title_headline_17)
+                    .foregroundColor(Color.white)
+                    .padding(.leading, 16)
+                    .frame(height: 17)
+                    .padding(.top, 7)
+    
+                List {
+                    ForEach(transactionsByDate.keys.sorted(by: >), id: \.self) { key in
```

Comment:
sorting strings that might contain "Today" "Yesterday" and numbered strings, this might not work as expected
for example:
`["Today", "01.12.24", "05.10.24"]`

Reply:
Also I think sorting should be done not here (this is poor UI layout code) but instead in init where we create `transactionsByDate` array

Reply:
fixed